Geometry_Engine/Query/IsContaining.cs

```diff
@@ -167,6 +169,154 @@ public static bool IsContaining(this Circle curve, List<Point> points, bool acce
 
         /***************************************************/
 
+        [Description("Checks if the colleciton of Points are all contained within the curve. If a single Point is outside the curve, the method will return false. Points not in the plane of the curve are deemed to be outside.")]
+        [Input("curve", "The Ellipse to check if it is containing all of the provided points.")]
+        [Input("points", "The points to check if they are all contained within the curve. If a single point is outside the curve or not in the plane of the curve the method will return false.")]
+        [Input("acceptOnEdge", "If true, points that are within the tolerance distance away from the curve are demmed to be inside it. If false, only points that are inside and not within tolerance distance away from the curve are deemed to be inside.")]
+        [Input("tolerance", "Distance tolerance to be used in the method. Points are deemed to be on the edge of the curve if they are within this distance from the curve.", typeof(Length))]
+        [Output("isContaining", "Returns true if all of the provided points are inside the curve.")]
+        public static bool IsContaining(this Ellipse curve, List<Point> points, bool acceptOnEdge = true, double tolerance = Tolerance.Distance)
+        {
+            if (curve.IsNull())
+                return false;
+
+
+            double minRad = Math.Min(curve.Radius1, curve.Radius2);
+
+            if (minRad < tolerance && !acceptOnEdge)
+            {
+                //If the minimum radius is lower than tolerance, the points will always be either on the edge or outside
+                return false;  
+            }
+
+            //Tranform the point to the local coordinates of the ellipse
+            //After tranformation can see it as the ellipse centred in the origin with first axis long global x and second axis along global y
+            Cartesian coordinateSystem = Create.CartesianCoordinateSystem(curve.Centre, curve.Axis1, curve.Axis2);
+            TransformMatrix transform = Create.OrientationMatrixLocalToGlobal(coordinateSystem);
```

Comment:
Again, is it only me to whom the name sounds quite opposite to what it actually does? 🙈

Reply:
Again, as above. I think the name actually is correct, when looking at it from the perspective of the ellipse, but does sound weirder when applying that same transformation to some other object!